### ISSUE TITLE
Removing specific chunks from training by read_id and mod position

### ISSUE THREAD

**Issue by andrewgalbraith21** [nanoporetech/remora#98]
Hello!

I am training a model to detect 8-Oxo-G overall training has gone well but we have reason to believe our modified OxoG set is not 100% pure in the desired context - we suspect experimental purity is closer to 95-99%. To deal with this, we want to remove chunks from the modified training set that are likely unmodified. To do this, I thought about collecting and extracting the read and signal position for all chunks with p(mod) < 0.25 (ML < 63) after running remora model infer. This is not hard to do but I am uncertain about how I can actually filter out these read and location specified chunks from the training.npz file. I have overviewed the source code but my understanding of python code is limited. Please let me know if there is a straightforward way this could be done thanks!

**Comment by marcus1487** [comment]
I can't think of a particularly easy way to do this with the current tools available. I would suggest having a look at [this bit of the code](https://github.com/nanoporetech/remora/blob/master/src/remora/prepare_train_data.py#L46). The io_read object can be found [here](https://github.com/nanoporetech/remora/blob/master/src/remora/io.py#L1363) and should contain all the information you need to filter the selected sites per read.

In terms of getting the information from the command line to this point in the code, you'd have to write your own parser and hook the code in through the parsers.py module.

For a more forward looking response, we are working on a major re-write of the remora datasets. As part of this it will be much easier to add arbitrary information to the dataset and later use this data to filter the chunks at training or to create a new dataset. This is still going to involve a bit more effort, but we can look into getting reference position into the remora dataset be default in relatively short order for this type of application.

**Comment by marcus1487** [comment]
I was thinking about this on the dataset side, but we do actually have an option in the `remora model train` command that might be a good replacement. The `--high-conf-incorrect-thr-frac` option takes two arguments and removes chunks (before the backwards pass through the model) which the current model confidently predict as a different label from the one in the dataset. So this is more of a dynamic approach from the one you are proposing, but hopefully might produce the intended results.

From the docs these two arguments are "1.) Threshold value of what to consider a high confidence predicition. Based on the softmax output. 2.) Fraction (of the batch size) of highly confident incorrect predictions to filter during training. Filtering up to this value, but might be lower." Let me know if this might suffice for your needs or if you have any questions about this option.

**Comment by andrewgalbraith21** [comment]
Hello Marcus,

Thanks so much for the advice! The --high-conf-incorrect-thr-frac sounds like an easier implementation than hard coding the dataset so I'll try that for now!
